app/models/spree/subscription.rb

```diff
@@ -127,6 +127,17 @@ def send_prior_notification
       end
     end
 
+    def cancel_stripe_subscription
+      begin
+        stripe_subscription = Stripe::Subscription.retrieve(self.stripe_subscription_id)
+        stripe_subscription.cancel
+      rescue Stripe::StripeError => e
+        Rails.logger.error "Stripe error while canceling subscription: #{e.message}"
+      rescue => e
+        Rails.logger.error "General error while canceling subscription: #{e.message}"
```

Comment:
your code wrongly falls in this exception if stripe subscription was deleted and not found and then line 133 will try to call cancel on nil. this will raise ruby undefined method error but actually error was from stripe subscription not available.